internal/db/profile_selector_scan.go

```diff
@@ -37,7 +38,11 @@ func (s *ProfileSelector) Scan(value interface{}) error {
 	str = strings.TrimSuffix(str, ")")
 
 	// Split the string by commas to get the individual field values
-	parts := strings.Split(str, ",")
+	cr := csv.NewReader(strings.NewReader(str))
+	parts, err := cr.Read()
+	if err != nil {
+		return fmt.Errorf("failed to scan SelectorInfo: %v", err)
+	}
```

Comment:
Do we need to update the reverse (store) code as well?

Do we have some unit tests to cover the round-trip as well as the serialized format?

Reply:
there is no serialization, really. The selectors are used as-is, but they are retrieved by aggregating the selectors table with the profiles table in https://github.com/stacklok/minder/blob/6f89a1de73ec7e42b1730951b645bbfb6c1387a0/database/query/profiles.sql#L68 which uses a SQL type defined at https://github.com/stacklok/minder/blob/6f89a1de73ec7e42b1730951b645bbfb6c1387a0/database/migrations/000072_profile_selectors_type.up.sql#L17 which we force through sqlc (https://github.com/stacklok/minder/blob/6f89a1de73ec7e42b1730951b645bbfb6c1387a0/sqlc.yaml#L31)

The code I modified is a Scan() to convert the Go type for the selectors defined in models.go to the sql type.

Reply:
Aha!  I'm slightly nervous about whether the Postgres `ARRAY_AGG` produces csv output, but this makes sense for  now.

Reply:
turns out that there is `pq.Array` which sounds like something I want to use. I'll take a look into using it - thanks for bringing up the concern!

Reply:
OK, I confused myself. `pq.Array` would be useful for parsing the whole array, but here we are parsing the `ROW`, not the array. I would still go with the csv approach, but we might want to consider using `LazyQuotes`. I'll do some more tests before we merge this PR.